packages/eyes-sdk-core/lib/sdk/CheckSettingsUtils.js

```diff
@@ -219,17 +230,17 @@ async function toScreenshotCheckSettings({checkSettings, context, screenshot}) {
           for (const element of elements) {
             const elementRegion = utils.geometry.withPadding(await element.getRegion(), reference.padding)
             const region = utils.geometry.offset(elementRegion, contextLocationInViewport)
-            referenceRegions.push(
-              utils.geometry.scale(
-                {
-                  x: Math.max(0, region.x - screenshot.region.x),
-                  y: Math.max(0, region.y - screenshot.region.y),
-                  width: region.width,
-                  height: region.height,
-                },
-                context.driver.viewportScale,
-              ),
+            const scaledRegion = utils.geometry.scale(
+              {
+                x: Math.max(0, region.x - screenshot.region.x),
+                y: Math.max(0, region.y - screenshot.region.y),
+                width: region.width,
+                height: region.height,
+              },
+              context.driver.viewportScale,
             )
+            scaledRegion.regionId = selector ? selector : stringifySelector(element)
```

Comment:
I see you saved `selector` from the region in the beginning and then use it to understand if you have to stringify the selector from an element. I would better always use a selector from an element.
```suggestion
            scaledRegion.regionId = element.selector && stringifySelector(element)
```

Reply:
Agreed, this was a hold-out from the initial implementation. Although I prefer a tell-don't-ask style of code. I would assume that stringifySelector (or whatever it calls) would work if element.selector exists otherwise return undefined. I'll commit a fix.

Reply:
Actually, it is my bad. What I recommended is not doing what I was imagining. 
it should be like this 
```
element.selector ? stringifySelector(element) : undefined
```